src/kpatch_process.c

```diff
@@ -1121,10 +1121,10 @@ kpatch_process_init(kpatch_process_t *proc,
 	list_init(&proc->vmaholes);
 	proc->num_objs = 0;
 
-	if (process_get_comm(proc))
-		goto out_unlock;
 	if (kpatch_coroutines_init(proc))
 		goto out_unlock;
+	if (process_get_comm(proc))
```

Comment:
May you please elaborate on this?

Reply:
If process already exited, process_get_comm returns error, proc->coro.coros list is not correctly initialized and process exits on exception at list iteration.